SU2_CFD/src/solvers/CSolver.cpp

```diff
@@ -4265,74 +4269,187 @@ void CSolver::LoadInletProfile(CGeometry **geometry,
+                    dist += pow(Inlet_Data[index+iDim] - Coord[iDim], 2);
+                    dist = sqrt(dist);
 
-            if (min_dist < tolerance) {
+                    /*--- Check is this is the closest point and store data if so. ---*/
 
-              solver[MESH_0][KIND_SOLVER]->SetInletAtVertex(Inlet_Values.data(), iMarker, iVertex);
+                    if (dist < min_dist) {
+                    min_dist = dist;
+                    for (iVar = 0; iVar < nColumns; iVar++)
+                        Inlet_Values[iVar] = Inlet_Data[index+iVar];
+                    }
 
-            } else {
-
-              unsigned long GlobalIndex = geometry[MESH_0]->node[iPoint]->GetGlobalIndex();
-              cout << "WARNING: Did not find a match between the points in the inlet file" << endl;
-              cout << "and point " << GlobalIndex;
-              cout << std::scientific;
-              cout << " at location: [" << Coord[0] << ", " << Coord[1];
-              if (nDim ==3) error_msg << ", " << Coord[2];
-              cout << "]" << endl;
-              cout << "Distance to closest point: " << min_dist << endl;
-              cout << "Current tolerance:         " << tolerance << endl;
-              cout << endl;
-              cout << "You can widen the tolerance for point matching by changing the value" << endl;
-              cout << "of the option INLET_MATCHING_TOLERANCE in your *.cfg file." << endl;
-              local_failure++;
+                }
+
+                /*--- If the diff is less than the tolerance, match the two.
+                We could modify this to simply use the nearest neighbor, or
+                eventually add something more elaborate here for interpolation. ---*/
+
+                if (min_dist < tolerance) {
+
+                    solver[MESH_0][KIND_SOLVER]->SetInletAtVertex(Inlet_Values.data(), iMarker, iVertex);
+
+                } else {
+
+                    unsigned long GlobalIndex = geometry[MESH_0]->node[iPoint]->GetGlobalIndex();
+                    cout << "WARNING: Did not find a match between the points in the inlet file" << endl;
+                    cout << "and point " << GlobalIndex;
+                    cout << std::scientific;
+                    cout << " at location: [" << Coord[0] << ", " << Coord[1];
+                    if (nDim ==3) error_msg << ", " << Coord[2];
+                    cout << "]" << endl;
+                    cout << "Distance to closest point: " << min_dist << endl;
+                    cout << "Current tolerance:         " << tolerance << endl;
+                    cout << endl;
+                    cout << "You can widen the tolerance for point matching by changing the value" << endl;
+                    cout << "of the option INLET_MATCHING_TOLERANCE in your *.cfg file." << endl;
+                    local_failure++;
+                    break;
+                }
+
+            }
+
+            else if(Interpolate == true){
```

Comment:
This routine is getting a bit long (and I did not help you by writing a fairly long version in the first place).. if you see obvious opportunities, break this out into some smaller helper functions. A good target to shoot for is about 40 lines of code per function (or no more than roughly one screen length, to help people like me who forget what they read once they scroll down 😉!)

Reply:
Are you talking about the whole LoadInletProfile subroutine or just the min_dist<tolerance error output part?

Reply:
and thanks alot for reviewing my code @economon

Reply:
I think the entire thing could be broken down further, if you see some opportunities.

Reply:
It can. But maybe we can leave it for sometime later. I don't want to mix up these two things.